src/main/scala/rps/model/ResponseBody.scala

```diff
@@ -0,0 +1,6 @@
+package rps.model
+
+case class ResponseBody(result: Option[Result],
+                        computerMove: Option[Move],
+                        ourMove: Option[Move],
+                        error: Option[String])
```

Comment:
When your data model has a lot of optionals fields, it's usually a code smell.

In this case, you're conflating two data models into one: the success case and the error case.

At this stage, it's not requested to handle the error via http, so you have two choices:

- using two classes and returning a proper 400 error instead
- model only the success case and throw an exception (which will result in a 500 error by default)

Reply:
Yes this smelled so much :) I also thought about separating the two but then didn't (purely out of lazyness.. 😪). I'll go with the first option 👍